actions/latest_testflight_version.rb

```diff
@@ -15,7 +15,10 @@ def self.run(params)
         if app.nil?
           UI.abort_with_message! "The application with bundle ID '#{params[:bundle_id]}' is not yet created in iTunes Connect."
         end
-        app.all_build_train_numbers.max || params[:initial_version_number]
+        if app.all_build_train_numbers.empty?
+          return params[:initial_version_number]
+        end
+        app.all_build_train_numbers.map { |v| Gem::Version.new(v) }.max.to_s
```

Comment:
If you look in PR #19 something similar is done, but we get the last version directly and instead of to_s we use the version property. I prefer that.
Although to that one we could add the nil validation and return initial_version_number if not.

Reply:
What do you mean by "the version property"?

I'm using `to_s` because I have a `Gem::Version` object, and I want to return a `String`.

I can change it to be 

`app.all_build_train_numbers.map { |v| Gem::Version.new(v) }.max.to_s  || params[:initial_version_number]`

Does this look better for you?

Reply:
Oh, it's not in that PR. What we did it last time.
`app.lastest_version.version || params[:initial_version_number]`

Reply:
OK yes, but that is not equivalent to this.

`latest_version` is based on the version exposed in app store. You can see them here:

`1.9.10`:
<img width="672" alt="screen shot 2018-01-15 at 3 04 02 pm" src="https://user-images.githubusercontent.com/287281/34955947-95d75c20-fa05-11e7-9c1f-785fe07ef6fd.png">

`1.0`
<img width="656" alt="screen shot 2018-01-15 at 3 04 43 pm" src="https://user-images.githubusercontent.com/287281/34955957-9ffc1fd8-fa05-11e7-9509-2fe54a4f50c2.png">

`all_build_train_numbers ` is based on the versions uploaded to itunes connect, they're independent from the versions uploaded to app store. You can see them here:

`1.1`:
<img width="669" alt="screen shot 2018-01-15 at 3 04 55 pm" src="https://user-images.githubusercontent.com/287281/34955988-cdd4971e-fa05-11e7-9853-c26e2f10e3af.png">

Reply:
Oh okay. I've been reading to understand. I get it now. And I guess that instead of asking for the build train numbers and then converting them, we can just ask for the objects: `build_trains` (hash) and then just get the values which are TrainBuild (keys are the version numbers).

Reply:
OK, didn't know that. I will give it a try. As long as I can order them by date or something to get the higher, it should be useful. I'll let you know.

Reply:
Good Find , this seems like a good solution

Reply:
I think this line could be replaced by: `app.build_trains.values.max.version_string`. Could you try it out @jcaracciolo ?

Reply:
Just tried it out and Spaceship::TestFlight::Build can't be compared, so we'd have the same problem. So like this is okay ! :D